jgrapht-core/src/main/java/org/jgrapht/alg/cycle/AhujaOrlinSharmaCyclicExchangeLocalAugmentation.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * (C) Copyright 2003-2018, by Christoph Grüne and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+
+import java.util.*;
+
+/**
+ * Implementation of an algorithm for the local augmentation problem for the cyclic exchange neighborhood,
+ * i.e. it finds subset-disjoint negative cycles, based on the
+ * <a href ="http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.15.6758&rep=rep1&type=pdf">paper</a>
```

Comment:
Please follow the citation conventions in

https://github.com/jgrapht/jgrapht/wiki/How-to-write-documentation

This is preferred over a link, since links may break at any time.